plugins/module_utils/network/sonic/config/ntp/ntp.py

```diff
@@ -0,0 +1,550 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2022 Dell Inc. or its subsidiaries. All Rights Reserved
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The sonic_ntp class
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import re
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.cfg.base import (
+    ConfigBase,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    to_list,
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.facts.facts import Facts
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.sonic import (
+    to_request,
+    edit_config
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.utils import (
+    get_diff,
+    update_states,
+    normalize_interface_name,
+    normalize_interface_name_list
+)
+from ansible.module_utils.connection import ConnectionError
+
+PATCH = 'PATCH'
+DELETE = 'DELETE'
+
+TEST_KEYS_ADD = [
+    {
+        "vrf": "", "enable_ntp_auth": "",
```

Comment:
The purpose of the "TEST_KEYS" is to provide the keys that uniquely identify each "list" item that is part of the resource module argument hierarchy (not all of the keys for each list item; only the keys that identify the item).

There is usually only one set of TEST_KEYS (not separate ones for ADD and DEL). 

The format of each dictionary in the list of dictionaries should be:

{"<list_name>":  {"<first_key_name>": "", "<second_key_name>": "", ... }},

In this case, it looks like the required key definitions are present, but there are also extra entries added here that will increase search time during diff calculation.

Reply:
It is neded when "have" has "ntp server x.x.x.x minpoll 5 maxpoll 8", then "want" has "ntp server x.x.x.x minpoll 8 maxpoll 10"

Reply:
After identifying the appropriate list or config element using the TEST_KEYS value or values that uniquely identify a "target" element (in a list or in the config), the "get_diff" function compares the other key/value items to determine if a change is required.

For the example cited, the server IP address alone is sufficient to identify the entry. Once it has been identified, the other key/value pairs (including minpoll and maxpoll are checked to see if they differ between the "have" and "want" dictionaries.

Also, the unique identifying keys for a given item are the same regardless of whether the operation is an ADD or DEL.

Reply:
Changed

Reply:
Verified.